### DISCUSSION TITLE
[Feature] Support .CR2 file format

### DISCUSSION THREAD

**Discussion by esmondmissen** [immich-app/immich#1672] [Feature Request]
**Describe the bug**
Canons image format .cr2 seem to fail to upload

**To Reproduce**
1. Add a CR2 file to your photo library
2. Open Immich
3. Try to run backup

**Expected behavior**
Upload and save .cr2 image to immich

**Screenshots**

**System**
- Phone OS: iPhone 12 pro iOS 16
- Server Version: 1.27.0
- Mobile App Version: 1.26.0 build.51

**Comment by alextran1502** [comment]
Can you help including a test file?

**Comment by esmondmissen** [comment]
[cr2.zip](https://github.com/immich-app/immich/files/9508063/cr2.zip)
Took me too long to work out how to attach a file on github. Here you go.

**Comment by anderiv** [comment]
I've just ran into this also - would love to see CR2 support! Thanks!

**Comment by x-real-ip** [comment]
I would also like to see CR2 file support :)

**Comment by anderiv** [comment]
As a stopgap before full CR2 support, it would be nice to have the app fail gracefully on CR2 files, or just exclude incompatible filetypes altogether. As it currently is, the presence of CR2 files in my library (which are a small minority of files present) is preventing the proper backup of the remaining files.

Thank you!

**Reply by jckmnhn** [reply]
The same is happening with me, except I have 150+ CR2 files. It attempts to go through each file but fails, preventing the backup of my remaining files.

**Reply by makzdot** [reply]
This! Have the same issue.

**Reply by deafmute1** [reply]
Whilst this might be quite graceful functionality, it seems like excluding files with a specific extension is something that the end user can do quite trivially. For example` immich upload $(find . -not -name "*.cr2")`.

**Comment by uhthomas** [comment]
Resolved by #2834.
